sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -184,9 +184,16 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper wit
                   // scalastyle:on
                   val newOutput = scan.readSchema().toAttributes
                   assert(newOutput.length == groupingExpressions.length + finalAggregates.length)
+                  val groupByExprToOutputOrdinal = mutable.HashMap.empty[Expression, Int]
+                  var ordinal = 0
                   val groupAttrs = normalizedGroupingExpressions.zip(newOutput).map {
                     case (a: Attribute, b: Attribute) => b.withExprId(a.exprId)
-                    case (_, b) => b
+                    case (expr, b) =>
+                      if (!groupByExprToOutputOrdinal.contains(expr.canonicalized)) {
+                        groupByExprToOutputOrdinal(expr.canonicalized) = ordinal
+                        ordinal += 1
```

Comment:
this looks incorrect. We use the ordinal to access `groupAttrs`, but here we skip incrementing `ordinal` if group by expression is an attribute. I think we should use `normalizedGroupingExpressions.zip(newOutput).zipWithIndex`.

BTW, do we have a test for mixed group by attr and group by expr?

Reply:
Thank you. I forgot the case.